docs/cookbook/adding-a-new-elasticsearch-index.md

```diff
@@ -0,0 +1,294 @@
+
+### Exporting via cron
+
+We may automate the export process via CronModule which is super easy.
+All we need to achieve this goal is to create a new class `CategoryExportCronModule` which extends `AbstractExportCronModule`.
+
+The most important task here is to override parent constructor with changing a type hint of the first argument to our created index (`CategoryIndex`).
+
+```php
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportCronModule;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader;
+use Shopsys\FrameworkBundle\Component\Elasticsearch\IndexFacade;
+
+class CategoryExportCronModule extends AbstractExportCronModule
+{
+    /**
+     * @param \App\Model\Category\Elasticsearch\CategoryIndex $index
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexFacade $indexFacade
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinitionLoader $indexDefinitionLoader
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        CategoryIndex $index,
+        IndexFacade $indexFacade,
+        IndexDefinitionLoader $indexDefinitionLoader,
+        Domain $domain
+    ) {
+        parent::__construct($index, $indexFacade, $indexDefinitionLoader, $domain);
+    }
+}
+```
+
+Now if we have [crons](../introduction/cron.md) properly configured the new cron will be started automatically with others.
+Or you may want to [configure](../cookbook/working-with-multiple-cron-instances.md) this cron module with different timing
+
+### Implement partial update
+
+Sometimes we want to export data immediately after the original data are changed (hiding category, rename, ...).
+For this purpose we need to implement `CategoryIndex::getExportDataForIds()`, scheduler for queuing, and [subscriber](https://symfony.com/doc/current/event_dispatcher.html#creating-an-event-subscriber) for processing queue.
+
+Anywhere you want to add row to immediate export you may call `CategoryExportScheduler::scheduleRowIdForImmediateExport()` for adding a row into the queue.
+
+#### Scheduler
+
+Scheduler is used as a queue of IDs which we want to export. When we make any changes we may add an affected category ID into this queue and Subscriber will pick it up after request is done.
+
+Create class `CategoryExportScheduler` which extends `AbstractExportScheduler`. We do not need to override nor implement any method.
+
+```php
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportScheduler;
+
+class CategoryExportScheduler extends AbstractExportScheduler
+{
+}
+```
+
+#### Subscriber
+
+Create class `CategoryExportSubscriber` which extends `AbstractExportSubsriber`.
+Override its `__construct()`, and `getSubscribedEvents()`.
```

Comment:
```suggestion
Override its `__construct()` and `getSubscribedEvents()` methods.
```

Reply:
What about an Oxfords comma? 😄

Reply:
I'm glad you're thinking about it 👍 
But Oxford comma takes place in a list of three or more items https://www.lexico.com/definition/oxford_comma